src/opm/output/eclipse/AggregateMSWData.cpp

```diff
@@ -459,6 +477,48 @@ namespace {
             };
         }
 
+        template <class RSegArray>
+        void assignSpiralICDCharacteristics(const ::Opm::Segment&    segment,
+                                            const ::Opm::UnitSystem& usys,
+                                            const int                baseIndex,
+                                            RSegArray&               rSeg)
+        {
+            using Ix = ::Opm::RestartIO::Helpers::VectorItems::RSeg::index;
+            using M  = ::Opm::UnitSystem::measure;
+
+            const auto& sicd = segment.spiralICD();
+
+            // Convert to [Pressure]/([Rate] * [Rate)]--Rate at reservoir cond.
+            rSeg[baseIndex + Ix::DeviceBaseStrength] = usys.from_si(M::pressure,
+                usys.to_si(M::rate, usys.to_si(M::rate, sicd->strength())));
```

Comment:
I would have guessed that the innermost conversion should have been: `usys.from_si(M::rate, sicd->string())` - now it looks like the `sicd->strength()` is in deck units?

Reply:
I **believe** it is correct to use `to_si()` here, although I did have think long and hard about this.  It is decidedly subtle.  Feel free to check my logic.

We're trying to create a conversion from internal (i.e., strictly SI) units, but to an output unit that is a ratio of the form
```
     [Pressure]
 -----------------
  [Rate] * [Rate]
```
Since the `[Rate]` is in the denominator we need to divide (twice) with the conversion factor (SI Rate -> Output Rate).  Assume "METRIC" conventions for simplicity.  Then SI Rate -> Output Rate is the same as (1 / SI Time) -> (1 / Metric Time) which is (Metric Time) -> (SI Time) = multiply by 86,400.

In other words we somehow need to manufacture a conversion constant of
```
      1.0e-5
--------------------
  86,400 * 86,400
```
between the `sicd->strength()` in internal units&mdash; in general very large numbers in the order of `1.0e+14`&mdash;and the output item in deck units.  On the other hand, when implementing conversions that don't already exist in the `UnitSystem::measure` by chaining together conversion calls we only really get to use multiplication as a basic operation.  That's why `.to_si()` effectively plays the role of "`* 1 / .from_si()`".

Does that make sense?

Reply:
> Does that make sense?

yes it makes sense; seems to correct to me. But maybe  a better solution would be to actually add the dimension `Pressure / Rate^2` to the unit system?